src/login/AuthProviderBase.ts

```diff
@@ -48,7 +49,14 @@ export abstract class AuthProviderBase<TLoginResult> {
 		}
 
 		const nonce: string = randomBytes(16).toString('base64');
-		const { server, redirectPromise, codePromise } = createServer(nonce);
+		const { server, redirectPromise, codePromise, codeTimer } = createServer(context, nonce);
+
+		cancellationToken.onCancellationRequested(() => {
+			server.close();
```

Comment:
According to https://nodejs.org/api/net.html#serverclosecallback `.close()` is async, and takes a callback. I think it'd be safest to use this callback in case it takes a while to close or there is an error.

Reply:
Not sure, maybe we don't care 😅

Reply:
The callback argument is optional and it's safe to omit it but I do think it's worth tracking any errors that come up. We don't expect this call to fail and if it does we don't really care (because we don't need the server at that point anyway) but I think it's better to know about the error than not.